security/s2a/internal/crypter/aesgcm.go

```diff
@@ -0,0 +1,97 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package crypter
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"fmt"
+)
+
+// Supported key sizes in bytes.
+const (
+	AesGcmKeySize128 = 16
+	AesGcmKeySize256 = 32
+)
+
+// aesgcm is the struct that holds necessary information for S2A record.
+type aesgcm struct {
+	aead cipher.AEAD
+}
+
+// NewAESGCM creates an instance that uses aesgcm for S2A record. Note that the
```

Comment:
Can we change `an instance that uses aesgcm for S2A record` to `an aesgcm implementation of the S2AAeadCrypter interface`?

Reply:
How does `NewAESGCM creates an instance of an AES-GCM crypter of the S2AAeadCrypter interface.` sound? I'm hesitant about `NewAESGCM creates an aesgcm implementation of the S2AAeadCrypter interface` since we're creating an instance of the class rather than an implementation. WDYT?

Reply:
What about if we just left `NewAESGCM creates an AES-GCM crypter instance.`? (Having `of the S2AAeadCrypter interface` at the end doesn't read so well.)